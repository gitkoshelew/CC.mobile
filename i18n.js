import i18next from 'i18next';
import {initReactI18next} from 'react-i18next';
import en from './public/locales/en.json';
import ru from './public/locales/ru.json';

i18next.use(initReactI18next).init({
  compatibilityJSON: 'v3',
  lng: 'en',
  fallbackLng: 'en',
  resources: {
    en,
    ru,
  },
  react: {
    useSuspense: false,
  },
});

export default i18next;
