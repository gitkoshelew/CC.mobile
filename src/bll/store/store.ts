import {combineReducers, configureStore} from '@reduxjs/toolkit';
import {appReducer} from '../appReducer';
import {quizReducer} from '../quizReducer';
import {processReducer} from '@src/bll/processReducer';
import {resultReducer} from '@src/bll/resultReducer';
import {authReducer} from '@src/bll/authReducer';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';

const rootReducer = combineReducers({
  app: appReducer,
  authReducer,
  quizReducer,
  processReducer,
  resultReducer,
});
const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
};
const persistedReducer = persistReducer(persistConfig, rootReducer);
export const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});
export type RootStateT = ReturnType<typeof store.getState>;
export type AppDispatchT = typeof store.dispatch;

export default store;
