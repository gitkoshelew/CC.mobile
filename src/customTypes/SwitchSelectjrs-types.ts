export interface ISwitchSelectProps {
  onPress?: (value: string) => void;
  value?: string | number;
  type: TypeSwitchSelect;
  disabled?: boolean;
}

export enum TypeSwitchSelect {
  LEVEL = 'level',
  NUMBER = 'number',
  FILTER = 'filter',
  TYPE_ANSWER = 'typeAnswer',
}
