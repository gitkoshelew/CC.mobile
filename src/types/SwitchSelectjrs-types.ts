export interface ISwitchSelectProps {
  onPress?: (value: string) => void;
  type: 'level' | 'number';
}
