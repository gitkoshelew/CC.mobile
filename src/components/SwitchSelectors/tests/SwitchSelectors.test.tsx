import {fireEvent, render} from '@testing-library/react-native';
import {SwitchSelectors} from '../index';
import {TypeSwitchSelect} from '@customTypes/SwitchSelectjrs-types';
import store from '@src/bll/store/store';
import {Provider} from 'react-redux';
import i18n from '../../../../i18n';
import {I18nextProvider} from 'react-i18next';

test('Should render switchSelectors list', async () => {
  const {getByText, toJSON} = render(
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <SwitchSelectors type={TypeSwitchSelect.LEVEL} />
      </Provider>
    </I18nextProvider>,
  );
  const text = getByText(i18n.getResource('en', 'SwitchSelectors', 'level.0'));
  fireEvent.changeText(text);
  expect(toJSON()).toMatchSnapshot();
});
