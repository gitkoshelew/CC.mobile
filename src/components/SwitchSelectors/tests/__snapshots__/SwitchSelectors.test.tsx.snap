// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render switchSelectors list 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
>
  <View
    accessibilityLabel={null}
    style={
      [
        {
          "flexDirection": "row",
        },
        {},
      ]
    }
    testID={null}
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 50,
            "height": 40,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderColor": "#EAEAEA",
              "borderRadius": 50,
              "borderWidth": 1,
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  {},
                ]
              }
            >
              Easy
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "color": "#000000",
                    "fontSize": 14,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  {},
                ]
              }
            >
              Medium
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "color": "#000000",
                    "fontSize": 14,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  {},
                ]
              }
            >
              Hard
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
